pkg/segment/structs/segstructs.go

```diff
@@ -819,3 +821,113 @@ func CheckUnsupportedFunctions(post *QueryAggregators) error {
 
 	return nil
 }
+
+// GetBucketValueForGivenField returns the value of the bucket for the given field name.
+// It checks for the field in the statistic results and group by keys.
+// If the field is not found, it returns nil.
+// The first return value is the value of the field.
+// The second return value is the index of the field in the group by keys.
+// The third return value is a boolean indicating if the field was found in the Statistic results.
+// For Statistic results, the index is -1. and Bool is true.
+// For GroupBy keys, the index is the index of the field in the group by keys; given that the Bucket key is a List Type. and Bool is false.
+// Otherwise, the index is -1 and Bool is false.
+func (br *BucketResult) GetBucketValueForGivenField(fieldName string) (interface{}, int, bool) {
```

Comment:
@AndrewHess I would like to get your thoughts on this method and the below Set Method.

Reply:
Based on your feedback I will write the test cases for both of the functions and then we can merge this.

Reply:
Those functions look good to me

Reply:
Got it, thank you!